module/zfs/dsl_deadlist.c

```diff
@@ -122,6 +123,20 @@ dsl_deadlist_compare(const void *arg1, const void *arg2)
 	return (AVL_CMP(dle1->dle_mintxg, dle2->dle_mintxg));
 }
 
+static int
+dsl_deadlist_cache_compare(const void *arg1, const void *arg2)
+{
+	const dsl_deadlist_cache_entry_t *dlce1 = arg1;
+	const dsl_deadlist_cache_entry_t *dlce2 = arg2;
+
+	if (dlce1->dlce_mintxg < dlce2->dlce_mintxg)
+		return (-1);
+	else if (dlce1->dlce_mintxg > dlce2->dlce_mintxg)
+		return (+1);
+	else
+		return (0);
```

Comment:
This can be simplified and sped up nby switching to `AVL_CMP()`.

```c
static int
dsl_deadlist_cache_compare(const void *arg1, const void *arg2)
{
	const dsl_deadlist_cache_entry_t *dlce1 = arg1;
	const dsl_deadlist_cache_entry_t *dlce2 = arg2;

	return (AVL_CMP(dlce1->dlce_mintxg, dlce2->dlce_mintxg));
}
```